File: Javascript/Basic Hashing/hashing.js



//basic hashing in javascript;

// what is hashing it's just pre compute and fetching method




function main(){
  
    let n=7;
    let arr=[1,2,1,3,2,5,6];
    let q=5;
    let nums=[1,2,3,4,5];
    
    let hash=[0,0,0,0,0,0,0];
    
    for(let i=0;i<n;i++){
      hash[arr[i]]+=1;
    }
  
  while(q--){
    
    console.log(hash[nums[q]])
    console.log(nums[q],{q})
  }
    
    
  }
  
  main()

File: C++/Basic Hashing/hashing.cpp

#include <iostream>
using namespace std;

int main() 
{
    int n;
    std::cin >> n;
    int arr[n];
    
    for(int i=0;i<n;i++){
      cin>>arr[i];
    }
    int q;
    cin>>q;
    
    int hash[n+1]={0};
    for(int i=0;i<n;i++){
      hash[arr[i]]+=1;
    }
    
    while(q--){
      int nums;
      cin>>nums;
      std::cout << hash[nums] << std::endl;
    }
    return 0;
}